### PR TITLE
Token Holder upgrade to Capital Provider

### PR DESCRIPTION
**Pull request by amarinkovic**
Token Holder should be able to upgrade himself to a Capital Provider role, assuming there is an approval for it.

### CODE REVIEW

**Comment by melks** [comment]
Task linked: [CU-8693mwg5j Upgrade to capital provider role](https://app.clickup.com/t/8693mwg5j)

File: src/libs/LibAdmin.sol [hunk 1]
```diff
@@ -233,43 +232,35 @@ library LibAdmin {
...
-
         if (!isTokenHolder && !isCapitalProvider) {
             revert InvalidSelfOnboardRoleApproval(_role);
         }
 
-        if (s.selfOnboarding[_userAddress].entityId != 0 && s.selfOnboarding[_userAddress].roleId != 0) {
-            revert EntityOnboardingAlreadyApproved(_userAddress);
-        }
-
-        if (s.existingEntities[_entityId]) {
```
Comment (kevin-fruitful) [inline src/libs/LibAdmin.sol:252]: Is it okay to remove this check here? Shouldn't we ensure that the entityId that will be assigned to the user hasn't already been used?

Comment (amarinkovic) [reply src/libs/LibAdmin.sol:252]: The requirement is for the user to be able to transition into the Capital Provider role, even after onboarding as Token Holder and his entity being created already.

Comment (kevin-fruitful) [reply src/libs/LibAdmin.sol:252]: After discussion:
A user can be assigned an entity that already exists. If self onboarding is approved with an entity that exists, then an entity isn't created. But if its an entity that doesn't exist then one is created.

**Review by kevin-fruitful — APPROVED** [review]

**Pull request merged**